core/cmd/hoverfly/main.go

```diff
@@ -283,6 +282,8 @@ func main() {
 		requestCache = nil
 	}
 
+	cfg.DisableProxyAuthoriation = *disableProxyAuth
```

Comment:
It's confusing to have this as a boolean. There are three options `proxyAuth` or `hoverflyHeader` or `disabled`. I think in retrospect that the default should be `disabled`. It's a lot simpler than multiple combinations of different options. For example, if you disable authentication it doesn't make sense to be able to configure whether we are using a header or not, as it's turned off.